### PR TITLE
feat(frontend): support user manager in frontend

### LINKED ISSUE #2944: feat: store user account info in MetaStore and provide related RPC (including notification support).

### PR DESCRIPTION
**Pull request by yezizp2012**
## What's changed and what's your intention?

In this PR, we support user create and drop in Frontend: 
```sql
CREATE USER name [ [ WITH ] option [ ... ] ]

where option can be:

      SUPERUSER | NOSUPERUSER
    | CREATEDB | NOCREATEDB
    | LOGIN | NOLOGIN
    | [ ENCRYPTED ] PASSWORD 'password' | PASSWORD NULL
```
```sql
DROP USER [ IF EXISTS ] name [, ...]
```

For next step, we should support grant/revoke privilege, add login verification and privilege check in frontend.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
Resolve #2944 #2945

### CODE REVIEW

File: src/frontend/src/user/root_user.rs [hunk 1]
```diff
@@ -0,0 +1,81 @@
+// Copyright 2022 Singularity Data
```
Comment (fuyufjh) [inline src/frontend/src/user/root_user.rs:1]: Should the file name be `user_manager.rs`?

File: src/frontend/Cargo.toml [hunk 2]
```diff
@@ -24,6 +24,7 @@ lazy_static = "1"
 log = "0.4"
 madsim = "=0.2.0-alpha.3"
 maplit = "1"
+md5 = "0.7.0"
```
Comment (fuyufjh) [inline src/frontend/Cargo.toml:27]:
```suggestion
md5 = "0.7"
```

Comment (fuyufjh) [inline src/frontend/Cargo.toml:27]:
```suggestion
md5 = "0.7"
```

File: src/frontend/Cargo.toml [hunk 3]
```diff
@@ -40,6 +41,7 @@ risingwave_source = { path = "../source" }
 risingwave_sqlparser = { path = "../sqlparser" }
 risingwave_storage = { path = "../storage" }
 serde_json = "1"
+sha2 = "0.10.2"
```
Comment (fuyufjh) [inline src/frontend/Cargo.toml:44]:
```suggestion
sha2 = "0.10"
```

Comment (fuyufjh) [inline src/frontend/Cargo.toml:44]:
```suggestion
sha2 = "0.10"
```

File: src/frontend/src/observer/observer_manager.rs [hunk 4]
```diff
@@ -40,18 +42,23 @@ pub(crate) struct ObserverManager {
...
     catalog_updated_tx: Sender<CatalogVersion>,
+    user_info_manager: Arc<RwLock<UserInfoManager>>,
+    user_info_updated_tx: Sender<UserInfoVersion>,
     hummock_snapshot_manager: HummockSnapshotManagerRef,
 }
 
 const RE_SUBSCRIBE_RETRY_INTERVAL: Duration = Duration::from_millis(100);
 
 impl ObserverManager {
+    #[allow(clippy::too_many_arguments)]
```
Comment (fuyufjh) [inline src/frontend/src/observer/observer_manager.rs:53]: May consider using a struct to pass these arguments. Just for readability. :)

Comment (fuyufjh) [inline src/frontend/src/observer/observer_manager.rs:53]: May consider using a struct to pass these arguments. Just for readability. :)

**Review by fuyufjh — APPROVED** [review]
LGTM!

**Review by fuyufjh — COMMENTED** [review]
----

File: src/sqlparser/tests/sqlparser_common.rs [hunk 5]
```diff
@@ -2953,6 +2953,28 @@ fn parse_materialized_drop_view() {
     }
 }
 
+#[test]
+fn parse_create_user() {
```
Comment (neverchanje) [inline src/sqlparser/tests/sqlparser_common.rs:2957]: Pls add new tests in src/sqlparser/tests/testdata

File: src/frontend/Cargo.toml (hunk 2 above)
Comment (neverchanje) [inline src/frontend/Cargo.toml:27]:
```suggestion
md5 = "0.7"
```

File: src/frontend/Cargo.toml (hunk 3 above)
Comment (neverchanje) [inline src/frontend/Cargo.toml:44]:
```suggestion
sha2 = "0.10"
```

File: src/frontend/src/observer/observer_manager.rs (hunk 4 above)
Comment (yezizp2012) [reply src/frontend/src/observer/observer_manager.rs:53]: Can't find a suitable struct naming so far. 🤔 NTFS.

**Pull request merged**